### ISSUE TITLE
Bug: [prefer-readonly] does not report if a member's property is reassigned

### ISSUE THREAD

**Issue by matyasasynth** [typescript-eslint/typescript-eslint#6005]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.8.4&sourceType=module&code=MYGwhgzhAEAqCmEAuB5ARgK3sJ0DeAsAFDTQAOArmiAJbDQBuYIF8AXNAHYUC2a8AJwDcxAL7FioSDATJ8xUmQE0mSeNDXJ0GaAF4u8AO5xEqTNiQAKAJQiS5ZavWazGAEx6Dx2a4s27CuRUtPQuNhwMAPY0ACby9qRIABY0EAB0LtppTCzq+gCMAAx2pKQA9GUaKemZmG7ZzKyeRSXQ4kSiQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGYUACAXAngA4oDG0AlobgLQrzkB2uA9IdIgGaLTXsCGAEwD2DePnRRu0IdEhgAviHlA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### Repro Code

```typescript
class TestObject {
  public value: number;
}

class Test {
  private testObj = new TestObject();
  private testObj2 = new TestObject();

  public test(): void {
    this.testObj.value = 10;
    // this.testObj2.value = 10;
  }
}
```

### ESLint Config

```javascript
module.exports = {
  "rules": { "@typescript-eslint/prefer-readonly": "error" }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

testObj and testObj2 is never reassigned; both should be marked as `readonly`

### Actual Result

If we're setting a property on the testObj, eslint does not recognise it should be marked as `readonly`.
(Uncommenting line #7 should not make the eslint error go away.)

### Versions

| package | version |
| -- | -- |
| `@typescript-eslint/eslint-plugin` | `5.43.0` |
| `@typescript-eslint/parser` | `5.43.0` |
| `TypeScript` | `4.8.4` |
| `ESLint` | `8.15.0` |
| `node` | `web` |

**Comment by JoshuaKGoldberg** [comment]
> (Uncommenting line https://github.com/typescript-eslint/typescript-eslint/pull/7 should not make the eslint error go away.)

...err, yes, I see that it does? https://typescript-eslint.io/play/#ts=4.8.4&sourceType=module&code=MYGwhgzhAEAqCmEAuB5ARgK3sJ0DeAUNNAA4CuaIAlsNAG5ghnwBc0AdmQLZrwBOAbgIBfAgVCQYCZPiKk+VBknjRlydBmgBeDvADucRKkzYkACgCUQ4iQVKVa4xgBM23QelPTloXPKUaVSNLNjoAeyoAE1liYiQACyoIADpHDWSGJhUdAEYABmtYhKTUow1nDMZmN3zC0WEgA&eslintrc=N4KABGBEBOCuA2BTAzpAXGYUACAXAngA4oDG0AlobgLQrzkB2uA9IdIgGaLTXsCGAEwD2DePnRRu0IdEhgAviHlA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

I can't reproduce any bad behavior in the playground. Am I missing something?

**Comment by matyasasynth** [comment]
@JoshuaKGoldberg  The actual bad behaviour is the error disappearing :-) Sorry if I was unclear.

Simplified example here:
https://typescript-eslint.io/play/#ts=4.8.4&sourceType=module&code=MYGwhgzhAEAqCmEAuB5ARgK3sJ0DeAsAFDTQAOArmiAJbDQBuYIF8AXNAHYUC2a8AJwDcxAL7FioSDATJ8xUmQE0mSeNDXJ0GaAF4u8AO5xEqTNiQAKAJQiiC8lVr1NV6xwYB7GgBN5JUg0ACxoIADpXbTCmFnV9AEYABjtScSJRIA&eslintrc=N4KABGBEBOCuA2BTAzpAXGYUACAXAngA4oDG0AlobgLQrzkB2uA9IdIgGaLTXsCGAEwD2DePnRRu0IdEhgAviHlA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

We get an "All is ok!" from the linter, but we SHOULD get this error:
`"Member 'testObj' is never reassigned; mark it as `readonly`. 6:3 - 6:38"
`

**Comment by JoshuaKGoldberg** [comment]
Oh got it - that is a bug, yes! Thanks for the clarification.
